### PR TITLE
feat(ui): Add "Projects Affected" section in Incidents

### PR DESCRIPTION
**Pull request by billyvg**
List projects affected inside of an Incident.

With a 2500ms sleep:

https://cl.ly/1e5e3795c850

### CODE REVIEW

**Comment by billyvg** [comment]
@markstory 	Thoughts on an interface like this @markstory when we want to use an organization's projects?

File: src/sentry/static/sentry/app/utils/projects.jsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    // check if projects store has any items, if not we should wait
+    if (!projects.length) {
+      return;
+    }
+
+    this.loaded = true;
+
+    const [inStore, notInStore] = partition(slugs, slug =>
+      [].find(project => project.slug === slug)
```
Comment (billyvg) [inline src/sentry/static/sentry/app/utils/projects.jsx]: should be `projects.find` but this is a WIP

File: src/sentry/static/sentry/app/utils/projects.jsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    if (!this.fetchQueue.size) {
+      return;
+    }
+
+    this.setState({
+      fetching: true,
+    });
+
+    await new Promise(resolve => setTimeout(resolve, 2500));
```
Comment (billyvg) [inline src/sentry/static/sentry/app/utils/projects.jsx]: WIP

**Comment by markstory** [comment]
Showing a list of slugs, would require pagination for larger accounts. Is there a possibility of using a typeahead + multi-select for this? Like how we do conditions and fields on discover?

**Comment by billyvg** [comment]
@markstory So in this case we have a list of project slugs from the incident details, what we're doing here is looking into the store to fetch if it exists, otherwise query projects endpoint for a specific list of slugs. We do this so we can basically fetch the project platform (for the icon).

This would allow us to future proof when we do remove `projects` from org details.

We could, in addition, have an interface where if we don't supply a list of slugs, we query for a list of projects and supply the render prop with the list of projects + "has more" property as well.

**Comment by markstory** [comment]
@billyvg I think that is good approach. It will also work well with a typeahead solution as the store will become incrementally populated as results are loaded. We might need additional methods in the store to provide a typeahead searching though.

File: src/sentry/static/sentry/app/utils/projects.jsx [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+      });
+    }
+  };
+
+  render() {
+    const {slugs, children} = this.props;
+
+    return children({
+      // May not need to expose this?
+      initiallyLoaded: this.state.initiallyLoaded,
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/projects.jsx]: If we don't have a use-case for it right now I would err on exposing less over more in the beginning.

File: src/sentry/static/sentry/app/utils/projects.jsx [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+
+      // Calls API and searches for project, accepts a callback function with signature:
+      //
+      // fn(searchTerm, {append: bool})
+      onSearch: this.handleSearch,
+    });
+  }
+}
+
+export default withProjects(withApi(Projects));
```
Comment (markstory) [inline src/sentry/static/sentry/app/utils/projects.jsx]: If this component has `withProjects()` on it how likely will it be that we won't have all projects in the client?

Comment (billyvg) [reply src/sentry/static/sentry/app/utils/projects.jsx]: As it is now... very unlikely it does not have all projects.

File: src/sentry/static/sentry/app/utils/projects.jsx (hunk 3 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/utils/projects.jsx]: Removed.

**Review by markstory — APPROVED** [review]

**Pull request merged**